botocore/endpoint_provider.py

```diff
@@ -169,17 +169,20 @@ def get_attr(self, value, path):
         names indicates the one to the right is nested. The index will always occur at
         the end of the path.
 
-        :type value: dict or list
+        :type value: dict or tuple
         :type path: str
         :rtype: Any
         """
         for part in path.split("."):
+            if value is None:
+                return None
             match = GET_ATTR_RE.search(part)
             if match is not None:
                 name, index = match.groups()
                 index = int(index)
-                value = value.get(name)
-                if value is None or index >= len(value):
+                if name:
+                    value = value.get(name)
+                if index >= len(value):
                     return None
                 return value[index]
```

Comment:
This seems broken compared to our old behavior. We're setting `value = value.get(name)` which can be None. We then assume it's non-None in the index check and again index into `value` on line 187. Is there something else mitigating that issue?

Do we have a more comprehensive set of tests to make sure this function does what we want?